src/object.c

```diff
@@ -692,28 +692,35 @@ char *strEncoding(int encoding) {
 
 /* =========================== Memory introspection ========================== */
 
+static size_t sizeOfStringObject(robj *o) {
+    size_t strSize = 0;
+    if (o->encoding == OBJ_ENCODING_INT) {
+        strSize = sizeof(long long);
```

Comment:
sizeof(*o) seems better as it takes into account the robj and long long value stored in o->ptr.